dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/BeamSqlPrimitive.java

```diff
@@ -67,34 +67,34 @@ public T getValue() {
   }
 
   public long getLong() {
-    return (Long) getValue();
+    return Long.valueOf(getString());
```

Comment:
In my understanding, BeamSqlPrimitive is `strong` typed. By `strong` I mean, it has a `type` field, a `value` field, type should match with value, if you have a `long`, but you invoke `getDouble`, then this `type-value` thing does not make sense any more, we can just drop the `type` field.

I guess you do this because you need to use this `convenient` in `BeamSqlAbsExpression`, just move this logic into `BeamSqlAbsExpression`.